File: TDDMicroExercises.ProposedSolution/CSharp/TurnTicketDispenser/TicketDispenser.cs
namespace TDDMicroExercises.OneSolution.TurnTicketDispenser
{
    public class TicketDispenser
    {
        ITurnNumberSequence turnNumberSequence;

        public TicketDispenser() : this(TurnNumberSequence.GlobalTurnNumberSequence)
        {
        }


        public TicketDispenser(ITurnNumberSequence turnNumberSequence)
        {
            this.turnNumberSequence = turnNumberSequence;
        }

        public TurnTicket GetTurnTicket()
        {
            int newTurnNumber = turnNumberSequence.GetNextTurnNumber();
            TurnTicket newTurnTicket = new TurnTicket(newTurnNumber);

            return newTurnTicket;
        }
    }
}

File: TypeScript/turn-ticket-dispenser/ticket-dispenser.ts
import TurnNumberSequence from './turn-number-sequence';
import TurnTicket from './turn-ticket';

export default class TicketDispenser {

	public getTurnTicket() {
		const newTurnNumber = TurnNumberSequence.getNextTurnNumber();
		const newTurnTicket = new TurnTicket(newTurnNumber);

		return newTurnTicket;
	}

}
